plugins/woocommerce-admin/client/products/fills/product-form-tab-fills.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import { registerPlugin } from '@wordpress/plugins';
+import {
+	__experimentalWooProductTabItem as WooProductTabItem,
+	__experimentalWooProductSectionItem as WooProductSectionItem,
+	useFormContext,
+} from '@woocommerce/components';
+import { Product } from '@woocommerce/data';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { PricingSection } from '../sections/pricing-section';
+import { ProductInventorySection } from '../sections/product-inventory-section';
+import { ProductShippingSection } from '../sections/product-shipping-section';
+import { OptionsSection } from '../sections/options-section';
+import { ProductVariationsSection } from '../sections/product-variations-section';
+import { TAB_GENERAL_ID } from './constants';
+
+const Tabs = () => {
+	const { values: product } = useFormContext< Product >();
+	const tabPropData = useMemo(
+		() => ( {
+			general: {
+				name: 'general',
+				title: 'General',
+			},
+			pricing: {
+				name: 'pricing',
+				title: 'Pricing',
+				disabled: !! product?.variations?.length,
+			},
+			inventory: {
+				name: 'inventory',
+				title: 'Inventory',
+				disabled: !! product?.variations?.length,
+			},
+			shipping: {
+				name: 'shipping',
+				title: 'Shipping',
+				disabled: !! product?.variations?.length,
+			},
+			options: {
+				name: 'options',
+				title: 'Options',
+			},
+		} ),
+		[]
+	);
```

Comment:
Mostly used `useMemo` here to avoid a re-render of each nested tab item and it's children (given its at the top of the tree).
Although this still gets triggered on every form update because of its reliance on `useFormContext`, hopefully we can update this after the `Form` component optimizations.